### PR TITLE
Add Vocabulary.getMaxTokenType()

### LINKED ISSUE #904: Vocabulary interface missing an iterable or equivalent method
Depreciation of Recognizer.getTokenNames() presently leaves no equivalent or convenient method to iterate over the valid set of symbolic token names.  Since the underlying String[] is now private, cannot iterate based on size. 

Adding a Vocabulary.getSymbolicNames() method, returning a String[], would be helpful.

### PR DESCRIPTION
**Pull request by msteiger**
Fixes #904

### CODE REVIEW

File: runtime/Java/src/org/antlr/v4/runtime/Vocabulary.java [hunk 1]
```diff
@@ -85,7 +93,7 @@
 	 *
 	 * <ul>
 	 *  <li>Tokens created by lexer rules.</li>
-	 *  <li>Tokens defined in a {@code tokens{}} block in a lexer or parser
+	 *  <li>Tokens defined in a <code>tokens{}</code> block in a lexer or parser
```
Comment (msteiger) [inline runtime/Java/src/org/antlr/v4/runtime/Vocabulary.java]: Fixes invalid curly brackets in JavaDoc.

File: runtime/Java/src/org/antlr/v4/runtime/VocabularyImpl.java [hunk 2]
```diff
@@ -94,6 +96,8 @@ public VocabularyImpl(String[] literalNames, String[] symbolicNames, String[] di
 		this.literalNames = literalNames != null ? literalNames : EMPTY_NAMES;
 		this.symbolicNames = symbolicNames != null ? symbolicNames : EMPTY_NAMES;
 		this.displayNames = displayNames != null ? displayNames : EMPTY_NAMES;
+		this.maxTokenType = Math.max(this.displayNames.length,
```
Comment (msteiger) [inline runtime/Java/src/org/antlr/v4/runtime/VocabularyImpl.java]: This is very conservative as it even include display names. Pinging @sharwell for feedback on this one.

**Comment by parrt** [comment]
hi. have you added yourself to contributors.txt?

**Comment by msteiger** [comment]
I've added myself to the file. Thanks!

**Comment by parrt** [comment]
ok, i hope to do a 4.5.3 soon.

**Comment by parrt** [comment]
This isn't quite right I think. off-by-one:

```
		this.maxTokenType =
			Math.max(this.displayNames.length,
					 Math.max(this.literalNames.length, this.symbolicNames.length));
```

That is the length but max token type is less by 1. I.e., A grammar with two tokens:

```
a : A B ;
```

gets:

```
        private static final String[] _LITERAL_NAMES = {
        };
        private static final String[] _SYMBOLIC_NAMES = {
                null, "A", "B"
        };
```

Max length is 3 so max token type must be 1 less: 2.

**Comment by msteiger** [comment]
Sorry, my fault. You're absolutely right.

**Comment by mike-lischke** [comment]
If you only have empty names that - 1 will result in a -1 for maxTokenType. Is that intended or should this value be clipped to 0?

**Comment by parrt** [comment]
hmm...good question. Actually -1 since max 0 would indicate 0 was a valid token index like tokens[0].

**Comment by mike-lischke** [comment]
Ok, thanks.

**Comment by parrt** [comment]
A related note: https://github.com/antlr/antlr4/issues/1309 maxTokenType for the ATN is set improperly due to bug in Grammar.getMaxTokenType(). :(

**Pull request merged**